harness/determined/exec/harness.py

```diff
@@ -3,11 +3,13 @@
 import faulthandler
 import logging
 import sys
-from typing import Iterator
+from typing import Iterator, Optional, Dict
 
 import determined as det
-from determined import core, horovod, load
+from determined import core, horovod, load, pytorch
 from determined.common.api import analytics, certs
+from determined.pytorch import PyTorchTrial, PyTorchTrialController, PyTorchTrialContext
+from determined.pytorch._pytorch_trial import TrainUnit
```

Comment:
Not only are you importing the symbol directly, but you are importing it from a privately named submodule.  If you need `TrainUnit` exposed outside of `det.pytorch`, then the name must exist in the `det.pytorch` namespace.

The full RFC proposing/explaining our python code layout is [here](https://github.com/rb-determined-ai/pedl/blob/9bd2168ec23fbba866194f3d7d3cfc243d267100/python_code_layout.md).